### PR TITLE
Make a plant or robot diagram from (a subset of) a scenario

### PR DESCRIPTION
**Pull request by RussTedrake**
@nepfaff -- i need to land the "replace joints with weld joints" logic in Drake, but these are the extra api elements that I was intending.

- - -
This change is [Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/280)

### CODE REVIEW

File: manipulation/station.py
Comment (nepfaff) [inline manipulation/station.py:268]: What do you think about type hints for private functions/members? Typing could be helpful for others using that functionality in the future, but detailed docstrings seem less necessary.

File: manipulation/station.py
Comment (nepfaff) [inline manipulation/station.py:375]: typo: "replaced"

On that note, what kind of comments should we include in code reviews for manipulation? I'd say we want to develop fast here, so we shouldn't be picky about code style. Docstrings, type hints, and major inefficiencies are probably worth pointing out. What do you think?
It might be worth writing up a small guideline in the developer notes if we start collaborating more here (I would be happy to do this as well).

File: manipulation/station.py
Comment (nepfaff) [inline manipulation/station.py:360]: The return type is missing. Same for the docstring.

File: manipulation/station.py
Comment (nepfaff) [inline manipulation/station.py:412]: The return type is missing. Same for the docstring.

**Review by nepfaff — APPROVED** [review]
Looks great! This should enable a lot cleaner code and avoid bugs.

**Review by RussTedrake — COMMENTED** [review]
My original plan was to wait until the drake PR's merge, but now that i've touched up the docstrings, I'd like to merge this as is (with the new assert guarding the unimplemented behavior).  PTAL.

> *[Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/280#-:-NmwoapLBEQzgn5ln6mI:b-3hc8mx)* status: 0 of 6 files reviewed, 3 unresolved discussions (waiting on @nepfaff)

___
*[`manipulation/station.py` line 268 at r1](https://reviewable.io/reviews/RussTedrake/manipulation/280#gh-1436478135:-NmwgnxlBZkO5fQnkcCE:b-vee4wd) ([raw file](https://github.com/RussTedrake/manipulation/blob/8f568d689a4db2767bd6a5459dae594303c2dabd/manipulation/station.py#L268)):*
<details><summary><i>Previously, nepfaff (Nicholas Pfaff) wrote…</i></summary><blockquote>

What do you think about type hints for private functions/members? Typing could be helpful for others using that functionality in the future, but detailed docstrings seem less necessary. 
</blockquote></details>

Done.  Here and throughout.
___
*[`manipulation/station.py` line 360 at r1](https://reviewable.io/reviews/RussTedrake/manipulation/280#gh-1436486386:-NmwjcrL-fnEzjOm4FM-:bc0zidr) ([raw file](https://github.com/RussTedrake/manipulation/blob/8f568d689a4db2767bd6a5459dae594303c2dabd/manipulation/station.py#L360)):*
<details><summary><i>Previously, nepfaff (Nicholas Pfaff) wrote…</i></summary><blockquote>

The return type is missing. Same for the docstring.
</blockquote></details>

Done.  I actually hadn't included those anywhere.  I've added them throughout.
___
*[`manipulation/station.py` line 375 at r1](https://reviewable.io/reviews/RussTedrake/manipulation/280#gh-1436485991:-NmwjlmsCdMf-dz2eeHI:b-896fix) ([raw file](https://github.com/RussTedrake/manipulation/blob/8f568d689a4db2767bd6a5459dae594303c2dabd/manipulation/station.py#L375)):*
<details><summary><i>Previously, nepfaff (Nicholas Pfaff) wrote…</i></summary><blockquote>

typo: "replaced"

On that note, what kind of comments should we include in code reviews for manipulation? I'd say we want to develop fast here, so we shouldn't be picky about code style. Docstrings, type hints, and major inefficiencies are probably worth pointing out. What do you think?
It might be worth writing up a small guideline in the developer notes if we start collaborating more here (I would be happy to do this as well). 
</blockquote></details>

Done.
___
*[`manipulation/station.py` line 412 at r1](https://reviewable.io/reviews/RussTedrake/manipulation/280#gh-1436487274:-Nmwjn0Z9haMeQN2LyNE:b-896fix) ([raw file](https://github.com/RussTedrake/manipulation/blob/8f568d689a4db2767bd6a5459dae594303c2dabd/manipulation/station.py#L412)):*
<details><summary><i>Previously, nepfaff (Nicholas Pfaff) wrote…</i></summary><blockquote>

The return type is missing. Same for the docstring.
</blockquote></details>

Done.

**Review by nepfaff — COMMENTED** [review]
Sounds good!

> *[Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/280#-:-NmxAI9i2wYCd848RELa:brxp9sd)* status: 0 of 6 files reviewed, all discussions resolved

**Review by nepfaff — COMMENTED** [review]
:lgtm:

> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/RussTedrake/manipulation/280#-:-NmxB3490AaJdbAUR0Ww:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @RussTedrake)

**Pull request merged**
